### PR TITLE
Add ability to re-attach consuming callback

### PR DESCRIPTION
**Pull request by pinepain**
**Changes:**
 - add ability to re-attach consuming callback in `AMQPQueue::consume()`; 
 - add `AMQP_JUST_CONSUME` constant for that;
 - cleanup queue consumer tag when `AMQPQueue::cancel()` called;
 - add `AMQPQueue::getConsumerTag()` method.

All changes BC compatible, doesn't break current API, has no known performance impacts and documented in stubs. 

As a demonstration, test files can be used.

**Fixes:**
 - RabbitMQ's Direct reply-to, issue https://github.com/pdezwart/php-amqp/issues/155.

**Description:**

The main problem that have in php-amqp for now - `basic.consume` request and actual messages reading is one method, which leads to situation where we can't easily separate that two parts - consuming request and actual consuming. This is a legacy part and we even can't break it (we stick to semantic versioning, so no BC in minor release allowed).

I decided to add `AMQP_JUST_CONSUME` constant which prevent `AMQPQueue::consume()` from sending `basic.consume` request. This adds some magic and I'm not happy from such solution, but at this time this is minimal effective change that required. It solves problem and doesn't add significant complexity to logic and codebase.

**Example**:

```php
$conn = new AMQPConnection();
$conn->connect();

$channel  = new AMQPChannel($conn);
$exchange = new AMQPExchange($channel);

$q_reply_to = new AMQPQueue($channel);
$q_reply_to->setName('amq.rabbitmq.reply-to');
$q_reply_to->consume(null, AMQP_AUTOACK);

// running request-response logic

$q_reply_to->consume(function (AMQPEnvelope $message, AMQPQueue $queue) {
    echo $message->getBody() . ': ' . $message->getRoutingKey() . PHP_EOL;

    echo 'Received on ', $queue->getName(), ' queue', PHP_EOL;

    return false;
}, AMQP_JUST_CONSUME);
```

When `AMQP_JUST_CONSUME` constant passed as flag or it ORed with other  flag constants, any other flags are ignored (it is logical, because we don't send anything to broker).

### CODE REVIEW

File: stubs/AMQPQueue.php [hunk 1]
```diff
@@ -350,4 +362,14 @@ public function getChannel()
     public function getConnection()
     {
     }
+
+    /**
+     * Get latest consumer tag. If no consumer available or the latest on was canceled empty string will be returned.
```
Comment (lstrojny) [inline stubs/AMQPQueue.php]: Can we use null for the error case?

Comment (pinepain) [inline stubs/AMQPQueue.php]: Yeah, but it adds more lines and few extra checks for this, and `is_empty()` and in `if` cases empty string will acts the same as null. And for now in `var_dump()` output we use empty string if no consumer tag set. But if you think `null` is more reasonable return value I will make such trivial changes.

Comment (lstrojny) [inline stubs/AMQPQueue.php]: Let’s go with null then, yes

Comment (pinepain) [inline stubs/AMQPQueue.php]: I changed to `null` in `var_dump()` output and in `AMQPQueue::getConsumerTag()`.

**Comment by lstrojny** [comment]
Can you squash and add a changelog entry?

**Comment by pinepain** [comment]
Done.

**Comment by lstrojny** [comment]
Thank you!

**Comment by makasim** [comment]
Given this example [amqpqueue_consume_multiple.phpt](https://github.com/pdezwart/php-amqp/blob/master/tests/amqpqueue_consume_multiple.phpt) is it somehow possible to know the exact queue the message came from?

**Comment by pinepain** [comment]
Nope.

**Comment by makasim** [comment]
@pinepain there are possibilities to implement it. Could you please look at https://github.com/pdezwart/php-amqp/issues/281

**Pull request merged**